[Az.Synapse] Remove the unnecessary breaking change of parameter '-SparkConfigFilePath' for Spark Pool related cmdlets

yanjungao718 opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description
Remove the unnecessary breaking change of parameter '-SparkConfigFilePath' for 'New-AzSynaspeSparkPool' and 'Update-AzSynaspeSparkPool' cmdlets

<!-- Please add a brief description of the changes made in this PR. If you have an ongoing or finished cmdlet design, please paste the link below. -->

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

msJinLei reviewed (changes requested):
Please refer to the following example to suppress breaking change errors
```
suppress the breaking change errors in "tools/StaticAnalysis/Exceptions/${ModuleName}/BreakingChangeIssues.csv"
"AssemblyFileName","ClassName","Target","Severity","ProblemId","Description","Remediation"
"Microsoft.Azure.PowerShell.Cmdlets.Network.dll","Microsoft.Azure.Commands.Network.NewAzFirewallPolicyNatRuleCommand","New-AzFirewallPolicyNatRule","0","1050","The parameter set '__AllParameterSets' for cmdlet 'New-AzFirewallPolicyNatRule' has been removed.","Add parameter set '__AllParameterSets' back to cmdlet 'New-AzFirewallPolicyNatRule'."
```

Please provide the migration guide of StorageSync for this major release.

Please fill in the following table 
|cmdlet|breaking change|old usage|new usage| 
|-|-|-|-| 

Thanks

yanjungao718 commented:
<html>
<body>
<!--StartFragment--><p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; color: rgb(31, 35, 40); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Please fill in the following table</p>

| cmdle  | breaking change | old usage | new usage|
|--------------|----------------|-----------|---------|
| New-AzSynapseSparkPool |  **[-SparkConfigFilePath]** parameter is deprecated and replaced by **[-SparkConfiguration]** | New-AzSynapseSparkPool -WorkspaceName $wsname -Name $sparkpoolname -SparkConfigFilePath $path | New-AzSynapseSparkPool -WorkspaceName $wsname -Name $sparkpoolname  -SparkConfiguration $config |
| Update-AzSynapseSparkPool | **[-SparkConfigFilePath]** parameter is deprecated and replaced by **[-SparkConfiguration]**  | Update-AzSynapseSparkPool -WorkspaceName $wsname -Name $sparkpoolname -SparkConfigFilePath $path  | Update-AzSynapseSparkPool -WorkspaceName $wsname -Name $sparkpoolname  -SparkConfiguration $config |


<!--EndFragment-->
</body>
</html>
